simulator/infra/ports/timing.h

```diff
@@ -25,6 +25,7 @@ class Cycle : public boost::totally_ordered<Cycle>
         constexpr auto operator<( const Cycle& rhs) const { return value < rhs.value; }
 
         constexpr void inc() { ++value; }
+        constexpr void dec() { if ( value) --value; }
```

Comment:
Decrementing interface for Cycle should be avoided, it opens possibility to move hands of our watches counterclockwise